store/src/java-test/com/zimbra/cs/html/owasp/OwaspHtmlSanitizerTest.java

```diff
@@ -689,23 +686,40 @@ public void testBug102910() throws Exception {
     @Test
     public void testZCS7621() throws Exception {
         String html = "<div class=\"gmail\" style=\"display:none; width:0; overflow:hidden; float:left; max-height:0;\" align=\"center\">";
-        String result = new OwaspHtmlSanitizer(html, true, null).sanitize();
+        String result = new OwaspHtmlSanitizer(html, true, null).sanitize(false);
         Assert.assertTrue(result.contains("display"));
         Assert.assertTrue(result.contains("float"));
     }
 
     @Test
     public void testZCS7784() throws Exception {
         String html = "<img class=\"gmail\" style=\"display:none; width:0; overflow:hidden;\" src=\"https://localhost:8443/service/home/~/?auth=co&loc=en_US&id=285&part=2.2\" >";
-        String result = new OwaspHtmlSanitizer(html, true, null).sanitize();
+        String result = new OwaspHtmlSanitizer(html, true, null).sanitize(false);
         Assert.assertTrue(result.contains("style"));
     }
 
     @Test
     public void testZBUG1215() throws Exception {
         String html = "<div id=\"noticias\"><div class=\"bloque\">BLOQUESSS</div></div>";
-        String result = new OwaspHtmlSanitizer(html, true, null).sanitize();
+        String result = new OwaspHtmlSanitizer(html, true, null).sanitize(false);
         // check that the id and class attributes are not removed
         Assert.assertTrue(result.equals(html));
     }
+
+    @Test
+    public void testBugTSS18004() throws Exception {
+        String malformedHtml = "<HTML><BODY><p dir=\"ltr\" style=\"-webkit-text-size-adjust:auto;line-height:1.38;margin-top:0pt;margin-bottom:0pt;\">"
+                + "<span style=\"font-family:Arial;font-size:14.666666984558105px-webkit-text-size-adjust:auto;\"\">Hello h</span>"
+                + "<span style=\"font-size:11pt;font-family:Arial;font-variant-ligatures:normal;font-variant-east-asian:normal;font-variant-position:normal;vertical-align:baseline\"\">ave a nice day.</span>"
+                + "</p></BODY></HTML>";
+        String result = new OwaspHtmlSanitizer(malformedHtml, true, null).cleanMalformedHtml(malformedHtml, true);
```

Comment:
Do we have a test to check this is returning the expected result when running through full owasp sanitization?

Reply:
Used `cleanMalformedHtml(malformedHtml, true)` to avoid the insertion of html tags by the `JTidy`.